pep-0517.txt

```diff
@@ -166,48 +166,71 @@ The build backend object is expected to have attributes which provide
 some or all of the following hooks. The common ``config_settings``
 argument is described after the individual hooks::
 
-  def get_build_requires(config_settings):
+  def build_sdist(source_directory, sdist_directory, config_settings):
+      ...
+
+MUST build an unpacked sdist from the ``source_directory`` and place it into the
+specified ``sdist_directory``. If the ``source_directory`` is already an
+unpacked sdist (as opposed to a VCS source tree) then it SHOULD simply copy the
```

Comment:
Thinking about this a bit more, what I think we need is a definition of what possible cases there are and what they mean.

This hasn't been important so far in this PEP, because we've just treated a 'source tree' - a directory containing `pyproject.toml` - as the input for building a wheel. If we're going to specify a `build_sdist` step which may have different behaviour in different cases, we need to define what an sdist is, and what the different possible inputs for the `build_sdist` step are.

There are (at least) four kinds of input to pip which can result in it installing from source (names, sdist paths/urls, local directories, VCS URLs). In which of these cases will it:

a. Not call the `build_sdist` hook
b. Call it but be prepared to handle its failure by installing another way
c. Call it and fail if the hook fails

And that's just with pip; then there's the question of other tools which may use this API, and what they expect from it.